### DISCUSSION TITLE
Is the client object individual for all client connections?

### DISCUSSION THREAD

**Discussion by JS-Aibel** [zauberzeug/nicegui#847] [Q&A] [answered]
### Question

Can we use the client object to store “global” variables to be used throughout the application? That way we only need to pass around the client object. In a multiuser application, will all clients have their own client object?
Or is it any best practice way to work with global variables in NiceGUI?

In the example below, the label gets stored in the client object. The `changeText(client)` method can then change the text of the label by only passing in the client object. 

```py
from nicegui import ui, Client

@ui.page('/')
async def test(client: Client):
    client.text = ui.label("Test")

    ui.button("Change text", on_click=lambda: changeText(client))

def changeText(client):
    client.text.set_text("New text")

ui.run()

```

**Comment by falkoschindler** [comment, marked as answer]
Yes, that's possible. Some thoughts about it:

- You loose type information. E.g. in your example, the IDE doesn't know that `client.text` is a `ui.label`.
- When the client is destroyed (because the browser tab disconnects), all its attributes can be garbage-collected. That's good.
- Make sure to distinguish clients and [sessions](https://nicegui.io/documentation#sessions): The same user might open the page with two tabs. Both will share the same session object, but have different client objects.
- Note that the [auto-index page](https://nicegui.io/documentation#auto-index_page) has a single client object for all browser tabs.
- Also, different pages have different client objects.
- The discussion https://github.com/zauberzeug/nicegui/discussions/732 might be related.
